### PR TITLE
Fixes for API 34

### PR DESCRIPTION
**Pull request by joshhood**
Currently there are two crashes occuring when attempting to use the MockNavigationActivity on Android API 34:

`java.lang.SecurityException: com.mapbox.services.android.navigation.testapp: One of RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED should be specified when a receiver isn't being registered exclusively for system broadcasts`

`java.lang.SecurityException: Starting FGS with type location callerApp=ProcessRecord{a043266 10859:com.mapbox.services.android.navigation.testapp/u0a410} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_LOCATION] any of the permissions allOf=false [android.permission.ACCESS_COARSE_LOCATION, android.permission.ACCESS_FINE_LOCATION]  and the app must be in the eligible state/exemptions to access the foreground only permission`

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
     <uses-permission android:name="android.permission.VIBRATE" />
     <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
```
Comment (Fabi755) [inline app/src/main/AndroidManifest.xml:6]: Is the `FOREGROUND_SERVICE` needed, when we declare `FOREGROUND_SERVICE_LOCATION`?

Comment (joshhood) [reply app/src/main/AndroidManifest.xml:6]: Yes. I have jsut tested it again and without that permission the following crash occurs

`java.lang.SecurityException: Permission Denial: startForeground from pid=7011, uid=10410 requires android.permission.FOREGROUND_SERVICE`

Comment (Fabi755) [reply app/src/main/AndroidManifest.xml:6]: Okay! Thanks for check this out!

**Review by Fabi755 — APPROVED** [review]
Thanks for your fix 💙

**Review by boldtrn — APPROVED** [review]
Thanks for providing these changes @joshhood, would you mind adding a CHANGELOG entry?

**Comment by joshhood** [comment]
Hi, sorry for slow reply, have added entry to CHANGELOG now

**Comment by boldtrn** [comment]
Thanks a lot for the PR!

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationNotification.java [hunk 2]
```diff
@@ -151,7 +151,7 @@ private PendingIntent createPendingOpenIntent(Context context) {
 
   private void registerReceiver(Context context) {
     if (context != null) {
-      context.registerReceiver(endNavigationBtnReceiver, new IntentFilter(END_NAVIGATION_ACTION));
+      context.registerReceiver(endNavigationBtnReceiver, new IntentFilter(END_NAVIGATION_ACTION), Context.RECEIVER_NOT_EXPORTED);
```
Comment (Fabi755) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationNotification.java:154]: May bad, I seeing this too late. This call of [`registerReceiver`](https://developer.android.com/reference/android/content/Context#registerReceiver(android.content.BroadcastReceiver,%20android.content.IntentFilter,%20int)) is only available for API +26.

Comment (Fabi755) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationNotification.java:154]: I'm currently investigate some time in #94. And will fix this too on this way.

Comment (boldtrn) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationNotification.java:154]: Thanks for investigating @Fabi755

**Pull request merged**